components/shared/code/libs/lib_nvm.c

```diff
@@ -365,6 +445,7 @@ static void evaluateWriteRequired(lib_nvm_entry_t entryId)
         records[entryId].writeRequired = true;
     }
```

Comment:
I'm not in love with this idea of checking if writes are required from within the nvm lib... Shouldn't the application using the lib be the one to determine if a write is required? _maybe_ there's a case for double-checking in the lib that a write hasn't been requested with exactly the same data, but otherwise this seems fishy to me

Reply:
We could only run this on things like the record and cycle log.

Did it this way so that the application can have an "assurance" that when it updates ram, it will seamlessly save data in the worse case of a system reset. For set and forget values, the application can request a write when its done transforming the data and then be assured that at the next time its needed, it will be ready for use